trestle/tasks/cis_to_catalog.py

```diff
@@ -0,0 +1,291 @@
+        text1 = '  input-dir              = '
+        text2 = '(required) location to read the compliance-as-code profile files.'
+        logger.info(text1 + text2)
+        text1 = '  output-dir             = '
+        text2 = '(required) location to write the generated catalog.json file.'
+        logger.info(text1 + text2)
+        text1 = '  output-overwrite       = '
+        text2 = '(optional) true [default] or false; replace existing output when true.'
+        logger.info(text1 + text2)
+
+    def simulate(self) -> TaskOutcome:
+        """Provide a simulated outcome."""
+        return TaskOutcome('simulated-success')
+
+    def execute(self) -> TaskOutcome:
+        """Provide an actual outcome."""
+        try:
+            return self._execute()
+        except Exception:
+            logger.info(traceback.format_exc())
+            return TaskOutcome('failure')
+
+    def _execute(self) -> TaskOutcome:
+        """Wrap the execute for exception handling."""
+        if not self._config:
+            logger.error('config missing')
+            return TaskOutcome('failure')
+        try:
+            idir = self._config['input-dir']
+            odir = self._config['output-dir']
+        except KeyError as e:
+            logger.info(f'key {e.args[0]} missing')
+            return TaskOutcome('failure')
+        # verbosity
+        quiet = self._config.get('quiet', False)
+        verbose = not quiet
+        # output
+        overwrite = self._config.getboolean('output-overwrite', True)
+        opth = pathlib.Path(odir)
+        # insure output dir exists
+        opth.mkdir(exist_ok=True, parents=True)
+        # calculate output file name & check writability
+        oname = 'catalog.json'
+        ofile = opth / oname
+        if not overwrite and pathlib.Path(ofile).exists():
+            logger.error(f'output: {ofile} already exists')
+            return TaskOutcome('failure')
+        # metadata links (optional)
+        metadata_links = self._config.get('metadata-links')
+        # get list or <name>.profile files
+        filelist = self._get_filelist(idir)
+        if len(filelist) < 1:
+            logger.error(f'input: {idir} no .profile file found')
+            return TaskOutcome('failure')
+        # initialize node list
+        self._node_map = {}
+        # process files
+        for fp in filelist:
+            lines = self._get_content(fp)
+            self._parse(lines)
+        # get root nodes
+        root_nodes = self._get_root_nodes()
+        # groups and controls
+        root = Group(title='root', groups=[])
+        for node in root_nodes:
+            group = Group(title=f'{node.name} {node.description}')
+            root.groups.append(group)
+            depth = self._depth(node.name)
+            if depth == 3:
+                self._add_groups(group, node.name, depth)
+            if depth == 2:
+                self._add_controls(group, node.name, depth)
+        # metadata
```

Comment:
Is it possible controls might not be in a group and instead should be in a controls attribute of the catalog itself?

Reply:
So far, this has not been the case. If that changes we can adjust the code accordingly.